test/higherorderfns.jl

```diff
@@ -8,24 +8,35 @@ module HigherOrderFnsTests
 
 using Test
 using SparseArrays
+using SparseArrays: nonzeros
 using LinearAlgebra
 using Random
 include("forbidproperties.jl")
-
+function test_map_and_map!(A, alloc_tests)
+    # --> test map entry point
+    fA = Array(A)
+    shapeA = size(A)
+    @test map(sin, A) == sparse(map(sin, fA))
+    @test map(cos, A) == sparse(map(cos, fA))
+    # --> test map! entry point
+    fX = copy(fA); X = similar(A)
+    map!(sin, X, A); X = similar(A) # warmup for @allocated
+    map!(sin, X, A); X = similar(A) # warmup for @allocated
+    @test alloc_tests || @allocated(map!(sin, X, A)) == 0
```

Comment:
Is this meant to be
```suggestion
    alloc_tests && @test @allocated(map!(sin, X, A)) == 0
```
?

Reply:
does it really matter?

Reply:
it was mean to be the other way around